### PR TITLE
Add connection pool pre-warming tests

### PR DESCRIPTION
**Pull request by ean5533**
1. The first test just needed to be uncommented now that https://github.com/square/okhttp/pull/8348 is merged
2. A second test was added to exercise http2-specific functionality

### CODE REVIEW

**Review by yschimke — APPROVED** [review]
Nice, very readable.

File: okhttp/src/test/java/okhttp3/internal/connection/ConnectionPoolTest.kt [hunk 1]
```diff
@@ -195,54 +203,133 @@ class ConnectionPoolTest {
...
+    setPolicy(pool, address, ConnectionPool.AddressPolicy(2))
+    assertThat(pool.connectionCount()).isEqualTo(2)
+
+    // Connections are replaced if they idle out or are evicted from the pool
+    evictAllConnections(pool)
+    assertThat(pool.connectionCount()).isEqualTo(2)
+    forceConnectionsToExpire(pool, routePlanner, expireTime)
+    assertThat(pool.connectionCount()).isEqualTo(2)
+
+    // Excess connections aren't removed until they idle out, even if no longer needed
```
Comment (swankjesse) [inline okhttp/src/test/java/okhttp3/internal/connection/ConnectionPoolTest.kt:224]: Great comments here

File: okhttp/src/test/java/okhttp3/internal/connection/ConnectionPoolTest.kt [hunk 2]
```diff
@@ -195,54 +203,133 @@ class ConnectionPoolTest {
...
+    setPolicy(pool, address, ConnectionPool.AddressPolicy(1))
+    assertThat(pool.connectionCount()).isEqualTo(2)
+    forceConnectionsToExpire(pool, routePlanner, expireTime)
+    assertThat(pool.connectionCount()).isEqualTo(1)
+  }
+
+  @Test fun connectionPreWarmingHttp2() {
```
Comment (swankjesse) [inline okhttp/src/test/java/okhttp3/internal/connection/ConnectionPoolTest.kt:231]: This is so cool

**Review by swankjesse — APPROVED** [review]

**Comment by yschimke** [comment]
```
ConnectionPoolTest > connectionPreWarmingHttp2() FAILED
    org.opentest4j.AssertionFailedError: expected:<[2]> but was:<[1]>
        at app//okhttp3.internal.connection.ConnectionPoolTest.connectionPreWarmingHttp2(ConnectionPoolTest.kt:256)
```

This is still 50% flaky.

**Pull request merged**